Vulcan: Problem: Answers CTA last; related problems before device card on mobile

ianrohde opened the pull request:
## Do This

- Always display the Answers CTA last
- On views where the device card and related problems are at the bottom of the page, display the related problems ahead of the device card
  - Device card stays on top in the sidebar, visible at desktop widths

## CR/QA

This is a new grid layout, and the syntax has a bit of a learning curve. If there are any questions, feel free to ping me.

These changes hit all breakpoints.

<details>
<img width="1716" alt="Screenshot 2023-11-16 at 4 32 19 PM" src="https://github.com/iFixit/react-commerce/assets/1634505/598e4c30-4ac9-4793-98a6-5c1c2fe21902">

<img width="1716" alt="Screenshot 2023-11-16 at 4 33 25 PM" src="https://github.com/iFixit/react-commerce/assets/1634505/681fa4d3-7820-473d-8174-478b573f422d">

</details>

Closes https://github.com/iFixit/ifixit/issues/50724

sctice-ifixit commented:
Looks great! But the conclusion (additional resources in the screenshots) should still come before related problems. I'm sorry that wasn't clear from the issue.

dev_block :rose:

ianrohde commented:
Order changed in https://github.com/iFixit/react-commerce/pull/2109/commits/ccacfcce04792e9ef02647935b832c277ac9c1f1

un_dev_block 🟢

aburke07 commented on frontend/templates/troubleshooting/index.tsx:
```diff
@@ -930,7 +947,7 @@ function AnswersCTA({ answersUrl }: { answersUrl: string }) {
                '@media (max-width: 375px)': {
                   flexDirection: 'column',
-                  gap: '6px',
+                  gap: 1.5,
```
Not a blocker because this was already here, but I believe we are no longer wanting to use `gap`. Can you use `spacing` here instead?

ianrohde replied:
Fixed in https://github.com/iFixit/react-commerce/pull/2109/commits/7dda83ba444e33993c565f00c2ef87e64d612276 👍

aburke07 reviewed (dismissed):
CR :alien:

aburke07 reviewed (approved):
CR :alien:

ianrohde commented:
dev_bock 🔴 Mattia updated the design, so I'll need to match it

ianrohde commented:
Updated per design in https://github.com/iFixit/react-commerce/pull/2109/commits/c258a2fe12aa16be6b2964ce6e7f875193ddc00a

Fixed a 320px mobile bug in https://github.com/iFixit/react-commerce/pull/2109/commits/b0026982d6f3f2ebc27786f95ca82a03605f46cb

<details>
<summary>Pre-fix bug screenshot</summary>
<img width="1716" alt="Screenshot 2023-11-17 at 9 59 53 AM" src="https://github.com/iFixit/react-commerce/assets/1634505/83c495e5-4c34-4d53-b84d-371b5ff083a3">

</details>

un_dev_bock 🟢

hcutrone reviewed (dismissed):
CR 🌯

deltuh-vee commented:
QA 🎬 
* The Answers CTA is always at the bottom of the screen
* On views where the device card and related problems are at the bottom of the page, the related problems is placed ahead of the device card 

@ianrohde 
Be careful with your branch names. I think this pull is failing CI because of the semicolon in the auto-generated Govinor URL. In general, the characters in branch names should be characters that can be part of a valid URL.

aburke07 reviewed (approved):
CR :alien:

hcutrone reviewed (approved):
CR 🌯

ianrohde commented:
Since the failure is only in the Vercel preview, I am going to proceed with merge. I'll keep the branch names in mind moving forward.
